### DISCUSSION TITLE
How to create a user without being authenticated/signed in as admin?

### DISCUSSION THREAD

**Discussion by casper-hansen** [fastapi/full-stack-fastapi-template#1190] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
curl -X 'POST' \
  'http://127.0.0.1:8000/api/v1/users/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "email": "test@test.com",
  "is_active": true,
  "is_superuser": false,
  "full_name": "Casper",
  "password": "test1234"
}'
```

### Description

I have tried to use the backend API to create a user. How do you create a user without being authenticated as an admin?

![image](https://github.com/tiangolo/full-stack-fastapi-template/assets/27340033/76381e55-a6ed-4bbe-b8b9-6ebad9260495)

### Operating System

Linux, macOS

### Operating System Details

Running locally on macOS or Linux has been tried.

### Python Version

Python 3.10.13

**Comment by DaveCLowe** [comment]
By design, you can't. Only superusers can create new users.

**Reply by casper-hansen** [reply]
That is what I understood as well. My question is more about how we can modify the code to allow this? It seems everything is gated by an auth token, so calling any of the endpoints will give a 401 Unauthorized

**Comment by PhilippWu** [comment]
Maybe you want that a new User can Register by hisself?

For that i would recommend you to build new Endpoints with the Logic to Register as new User. Including Email approvement etc.

The current Endpoints are not for User Self Registration so it makes sense that only Superuser can create new Users.

**Reply by casper-hansen** [reply]
Yes, I want the users to register their own accounts. Are there any recommended way of implementing a signup for users that is still secure enough?

**Comment by alejsdev** [comment, marked as answer]
Hi there! while the frontend doesn't currently support this feature, we've already implemented the backend endpoint register_user. To enable user registration, simply set USERS_OPEN_REGISTRATION to True in the .env file. After making this change, remember to restart the Docker containers using docker-compose up -d to apply the new environment variables.

**Reply by casper-hansen** [reply]
That sounds great. How do you protect this endpoint against potential abuse/spam?

**Reply by alejsdev** [reply]
Hmm, that depends more on the specific application, but having email verification may help because this would require users to verify their email addresses before allowing full access to the app

**Comment by alejsdev** [comment]
I will update the docs to include this :smile: 
https://github.com/tiangolo/full-stack-fastapi-template/pull/1191
